Loud warnings when compiling percolator

```
[build] [965/981  97% :: 216.337] Building CXX object src/openms/CMakeFiles/OpenMS.dir/thirdparty/percolator/DataSet.cpp.o
[build] In file included from /Users/julianuspfeuffer/git/OpenMS/src/openms/thirdparty/percolator/DataSet.cpp:18:
[build] In file included from /Users/julianuspfeuffer/git/OpenMS/src/openms/thirdparty/percolator/DataSet.h:34:
[build] In file included from /Users/julianuspfeuffer/git/OpenMS/src/openms/thirdparty/percolator/Scores.h:37:
[build] /Users/julianuspfeuffer/git/OpenMS/src/openms/thirdparty/percolator/ScoreHolder.h:48:9: warning: field 'label' will be initialized after field 'pPSM' [-Wreorder-ctor]
[build]    48 |         label(LabelType::UNDEFINED),
[build]       |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~
[build]       |         pPSM(NULL)
[build]    49 |         pPSM(NULL) {}
[build]       |         ~~~~~~~~~~
[build]       |         label(LabelType::UNDEFINED)
[build] /Users/julianuspfeuffer/git/OpenMS/src/openms/thirdparty/percolator/ScoreHolder.h:51:45: warning: field 'label' will be initialized after field 'pPSM' [-Wreorder-ctor]
[build]    51 |       : score(s), q(0.0), pep(0.0), p(0.0), label(l), pPSM(psm) {}
[build]       |                                             ^~~~~~~~  ~~~~~~~~~
[build]       |                                             pPSM(psm) label(l)
[build] 2 warnings generated.
[build] [966/981  97% :: 216.536] Building CXX object src/openms/CMakeFiles/OpenMS.dir/thirdparty/percolator/CrossValidation.cpp.o
[build] In file included from /Users/julianuspfeuffer/git/OpenMS/src/openms/thirdparty/percolator/CrossValidation.cpp:18:
[build] In file included from /Users/julianuspfeuffer/git/OpenMS/src/openms/thirdparty/percolator/CrossValidation.h:25:
[build] In file included from /Users/julianuspfeuffer/git/OpenMS/src/openms/thirdparty/percolator/DataSet.h:34:
[build] In file included from /Users/julianuspfeuffer/git/OpenMS/src/openms/thirdparty/percolator/Scores.h:37:
[build] /Users/julianuspfeuffer/git/OpenMS/src/openms/thirdparty/percolator/ScoreHolder.h:48:9: warning: field 'label' will be initialized after field 'pPSM' [-Wreorder-ctor]
[build]    48 |         label(LabelType::UNDEFINED),
[build]       |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~
[build]       |         pPSM(NULL)
[build]    49 |         pPSM(NULL) {}
[build]       |         ~~~~~~~~~~
[build]       |         label(LabelType::UNDEFINED)
[build] /Users/julianuspfeuffer/git/OpenMS/src/openms/thirdparty/percolator/ScoreHolder.h:51:45: warning: field 'label' will be initialized after field 'pPSM' [-Wreorder-ctor]
[build]    51 |       : score(s), q(0.0), pep(0.0), p(0.0), label(l), pPSM(psm) {}
[build]       |                                             ^~~~~~~~  ~~~~~~~~~
[build]       |                                             pPSM(psm) label(l)
[build] /Users/julianuspfeuffer/git/OpenMS/src/openms/thirdparty/percolator/CrossValidation.cpp:55:7: warning: field 'trainBestPositive_' will be initialized after field 'numThreads_' [-Wreorder-ctor]
[build]    48 |       testFdr_(testFdr),
[build]       |       ~~~~~~~~~~~~~~~~~
[build]       |       numThreads_(numThreads)
[build]    49 |       selectionFdr_(selectionFdr),
[build]       |       ~~~~~~~~~~~~~~~~~~~~~~~~~~~
[build]       |       testFdr_(testFdr)
[build]    50 |       initialSelectionFdr_(initialSelectionFdr),
[build]       |       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
[build]       |       selectionFdr_(selectionFdr)
[build]    51 |       selectedCpos_(selectedCpos),
[build]       |       ~~~~~~~~~~~~~~~~~~~~~~~~~~~
[build]       |       initialSelectionFdr_(initialSelectionFdr)
[build]    52 |       selectedCneg_(selectedCneg),
[build]       |       ~~~~~~~~~~~~~~~~~~~~~~~~~~~
[build]       |       selectedCpos_(selectedCpos)
[build]    53 |       niter_(niter),
[build]       |       ~~~~~~~~~~~~~
[build]       |       selectedCneg_(selectedCneg)
[build]    54 |       nestedXvalBins_(nestedXvalBins),
[build]       |       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
[build]       |       niter_(niter)
[build]    55 |       trainBestPositive_(trainBestPositive),
[build]       |       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
[build]       |       nestedXvalBins_(nestedXvalBins)
[build]    56 |       numThreads_(numThreads),
[build]       |       ~~~~~~~~~~~~~~~~~~~~~~~
[build]       |       trainBestPositive_(trainBestPositive)
[build] 3 warnings generated.
[build] [967/981  97% :: 216.680] Building CXX object src/openms/CMakeFiles/OpenMS.dir/thirdparty/percolator/LogisticRegression.cpp.o
[build] [968/981  97% :: 216.743] Building CXX object src/openms/CMakeFiles/OpenMS.dir/thirdparty/percolator/PseudoRandom.cpp.o
[build] [969/981  97% :: 217.070] Building CXX object src/openms/CMakeFiles/OpenMS.dir/thirdparty/percolator/IsotonicPEP.cpp.o
[build] [970/981  97% :: 217.279] Building CXX object src/openms/CMakeFiles/OpenMS.dir/thirdparty/percolator/MyException.cpp.o
[build] [971/981  97% :: 217.416] Building CXX object src/openms/CMakeFiles/OpenMS.dir/thirdparty/percolator/NoNormalizer.cpp.o
[build] [972/981  97% :: 217.455] Building CXX object src/openms/CMakeFiles/OpenMS.dir/thirdparty/percolator/Numerical.cpp.o
[build] [973/981  97% :: 217.613] Building CXX object src/openms/CMakeFiles/OpenMS.dir/thirdparty/percolator/Normalizer.cpp.o
[build] [974/981  97% :: 217.841] Building CXX object src/openms/CMakeFiles/OpenMS.dir/thirdparty/percolator/PackedVector.cpp.o
[build] [975/981  98% :: 217.918] Building CXX object src/openms/CMakeFiles/OpenMS.dir/thirdparty/percolator/PackedMatrix.cpp.o
[build] /Users/julianuspfeuffer/git/OpenMS/src/openms/thirdparty/percolator/PackedMatrix.cpp:43:23: warning: variable 'trhs' is uninitialized when used within its own initialization [-Wuninitialized]
[build]    43 |   PackedMatrix trhs = trhs.packedTranspose(rhs);
[build]       |                ~~~~   ^~~~
[build] 1 warning generated.
[build] [976/981  98% :: 218.032] Building CXX object src/openms/CMakeFiles/OpenMS.dir/source/APPLICATIONS/OpenSwathBase.cpp.o
[build] [977/981  98% :: 218.199] Building CXX object src/openms/CMakeFiles/OpenMS.dir/thirdparty/percolator/PSMDescription.cpp.o
[build] [978/981  98% :: 218.340] Building CXX object src/openms/CMakeFiles/OpenMS.dir/thirdparty/percolator/ResultHolder.cpp.o
[build] [979/981  98% :: 218.762] Building CXX object src/openms/CMakeFiles/OpenMS.dir/thirdparty/percolator/PosteriorEstimator.cpp.o
[build] [980/981  98% :: 219.040] Building CXX object src/openms/CMakeFiles/OpenMS.dir/thirdparty/percolator/SanityCheck.cpp.o
[build] In file included from /Users/julianuspfeuffer/git/OpenMS/src/openms/thirdparty/percolator/SanityCheck.cpp:20:
[build] In file included from /Users/julianuspfeuffer/git/OpenMS/src/openms/thirdparty/percolator/DataSet.h:34:
[build] In file included from /Users/julianuspfeuffer/git/OpenMS/src/openms/thirdparty/percolator/Scores.h:37:
[build] /Users/julianuspfeuffer/git/OpenMS/src/openms/thirdparty/percolator/ScoreHolder.h:48:9: warning: field 'label' will be initialized after field 'pPSM' [-Wreorder-ctor]
[build]    48 |         label(LabelType::UNDEFINED),
[build]       |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~
[build]       |         pPSM(NULL)
[build]    49 |         pPSM(NULL) {}
[build]       |         ~~~~~~~~~~
[build]       |         label(LabelType::UNDEFINED)
[build] /Users/julianuspfeuffer/git/OpenMS/src/openms/thirdparty/percolator/ScoreHolder.h:51:45: warning: field 'label' will be initialized after field 'pPSM' [-Wreorder-ctor]
[build]    51 |       : score(s), q(0.0), pep(0.0), p(0.0), label(l), pPSM(psm) {}
[build]       |                                             ^~~~~~~~  ~~~~~~~~~
[build]       |                                             pPSM(psm) label(l)
[build] /Users/julianuspfeuffer/git/OpenMS/src/openms/thirdparty/percolator/SanityCheck.cpp:34:57: warning: field 'concatenatedSearch_' will be initialized after field 'test_fdr_' [-Wreorder-ctor]
[build]    34 |   initPositives_(0), pTestset_(NULL), pTrainset_(NULL), concatenatedSearch_(true),
[build]       |                      ~~~~~~~~~~~~~~~  ~~~~~~~~~~~~~~~~  ^~~~~~~~~~~~~~~~~~~~~~~~~
[build]       |                      test_fdr_(0.01)  initial_train_fdr_(0.01) pTestset_(NULL)
[build]    35 |   test_fdr_(0.01), initial_train_fdr_(0.01) {
[build]       |   ~~~~~~~~~~~~~~~  ~~~~~~~~~~~~~~~~~~~~~~~~
[build]       |   pTrainset_(NULL) concatenatedSearch_(true)
[build] 3 warnings generated.
[build] [980/981  98% :: 219.452] Building CXX object src/openms/CMakeFiles/OpenMS.dir/thirdparty/percolator/Set.cpp.o
[build] [980/981  98% :: 219.726] Building CXX object src/openms/CMakeFiles/OpenMS.dir/thirdparty/percolator/MonotoneRegressor.cpp.o
[build] [980/981  98% :: 219.774] Building CXX object src/openms/CMakeFiles/OpenMS.dir/thirdparty/percolator/Timer.cpp.o
[build] [980/981  98% :: 219.786] Building CXX object src/openms/CMakeFiles/OpenMS.dir/thirdparty/percolator/StdvNormalizer.cpp.o
[build] [980/981  99% :: 219.908] Building CXX object src/openms/CMakeFiles/OpenMS.dir/thirdparty/percolator/UniNormalizer.cpp.o
[build] [980/981  99% :: 220.057] Building CXX object src/openms/CMakeFiles/OpenMS.dir/source/APPLICATIONS/TOPPBase.cpp.o
[build] [980/981  99% :: 220.134] Building CXX object src/openms/CMakeFiles/OpenMS.dir/thirdparty/percolator/Vector.cpp.o
[build] [980/981  99% :: 220.294] Building CXX object src/openms/CMakeFiles/OpenMS.dir/thirdparty/percolator/InProcessSampler.cpp.o
[build] [980/981  99% :: 220.321] Building CXX object src/openms/CMakeFiles/OpenMS.dir/thirdparty/percolator/SetHandler.cpp.o
[build] In file included from /Users/julianuspfeuffer/git/OpenMS/src/openms/thirdparty/percolator/SetHandler.cpp:18:
[build] In file included from /Users/julianuspfeuffer/git/OpenMS/src/openms/thirdparty/percolator/SetHandler.h:37:
[build] In file included from /Users/julianuspfeuffer/git/OpenMS/src/openms/thirdparty/percolator/DataSet.h:34:
[build] In file included from /Users/julianuspfeuffer/git/OpenMS/src/openms/thirdparty/percolator/Scores.h:37:
[build] /Users/julianuspfeuffer/git/OpenMS/src/openms/thirdparty/percolator/ScoreHolder.h:48:9: warning: field 'label' will be initialized after field 'pPSM' [-Wreorder-ctor]
[build]    48 |         label(LabelType::UNDEFINED),
[build]       |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~
[build]       |         pPSM(NULL)
[build]    49 |         pPSM(NULL) {}
[build]       |         ~~~~~~~~~~
[build]       |         label(LabelType::UNDEFINED)
[build] /Users/julianuspfeuffer/git/OpenMS/src/openms/thirdparty/percolator/ScoreHolder.h:51:45: warning: field 'label' will be initialized after field 'pPSM' [-Wreorder-ctor]
[build]    51 |       : score(s), q(0.0), pep(0.0), p(0.0), label(l), pPSM(psm) {}
[build]       |                                             ^~~~~~~~  ~~~~~~~~~
[build]       |                                             pPSM(psm) label(l)
[build] In file included from /Users/julianuspfeuffer/git/OpenMS/src/openms/thirdparty/percolator/SetHandler.cpp:18:
[build] /Users/julianuspfeuffer/git/OpenMS/src/openms/thirdparty/percolator/SetHandler.h:95:13: warning: 'const' type qualifier on return type has no effect [-Wignored-qualifiers]
[build]    95 |   LabelType const getLabel(int setPos);
[build]       |             ^~~~~
[build] /Users/julianuspfeuffer/git/OpenMS/src/openms/thirdparty/percolator/SetHandler.cpp:79:11: warning: 'const' type qualifier on return type has no effect [-Wignored-qualifiers]
[build]    79 | LabelType const SetHandler::getLabel(int setPos) {
[build]       |           ^~~~~
[build] /Users/julianuspfeuffer/git/OpenMS/src/openms/thirdparty/percolator/SetHandler.cpp:94:27: warning: variable 'hasRt' set but not used [-Wunused-but-set-variable]
[build]    94 |   bool hasScannr = false, hasRt = false, hasDm = false;
[build]       |                           ^
[build] 5 warnings generated.
[build] [980/981  99% :: 220.526] Building CXX object src/openms/CMakeFiles/OpenMS.dir/thirdparty/percolator/ScoreHolder.cpp.o
[build] In file included from /Users/julianuspfeuffer/git/OpenMS/src/openms/thirdparty/percolator/ScoreHolder.cpp:18:
[build] /Users/julianuspfeuffer/git/OpenMS/src/openms/thirdparty/percolator/ScoreHolder.h:48:9: warning: field 'label' will be initialized after field 'pPSM' [-Wreorder-ctor]
[build]    48 |         label(LabelType::UNDEFINED),
[build]       |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~
[build]       |         pPSM(NULL)
[build]    49 |         pPSM(NULL) {}
[build]       |         ~~~~~~~~~~
[build]       |         label(LabelType::UNDEFINED)
[build] /Users/julianuspfeuffer/git/OpenMS/src/openms/thirdparty/percolator/ScoreHolder.h:51:45: warning: field 'label' will be initialized after field 'pPSM' [-Wreorder-ctor]
[build]    51 |       : score(s), q(0.0), pep(0.0), p(0.0), label(l), pPSM(psm) {}
[build]       |                                             ^~~~~~~~  ~~~~~~~~~
[build]       |                                             pPSM(psm) label(l)
[build] 2 warnings generated.
[build] [980/981  99% :: 220.677] Building CXX object src/openms/CMakeFiles/OpenMS.dir/thirdparty/percolator/tron.cpp.o
[build] [980/981  99% :: 220.693] Building CXX object src/openms/CMakeFiles/OpenMS.dir/thirdparty/percolator/ssl.cpp.o
[build] [980/981  99% :: 221.138] Building CXX object src/openms/CMakeFiles/OpenMS.dir/thirdparty/percolator/Scores.cpp.o
[build] In file included from /Users/julianuspfeuffer/git/OpenMS/src/openms/thirdparty/percolator/Scores.cpp:31:
[build] In file included from /Users/julianuspfeuffer/git/OpenMS/src/openms/thirdparty/percolator/DataSet.h:34:
[build] In file included from /Users/julianuspfeuffer/git/OpenMS/src/openms/thirdparty/percolator/Scores.h:37:
[build] /Users/julianuspfeuffer/git/OpenMS/src/openms/thirdparty/percolator/ScoreHolder.h:48:9: warning: field 'label' will be initialized after field 'pPSM' [-Wreorder-ctor]
[build]    48 |         label(LabelType::UNDEFINED),
[build]       |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~
[build]       |         pPSM(NULL)
[build]    49 |         pPSM(NULL) {}
[build]       |         ~~~~~~~~~~
[build]       |         label(LabelType::UNDEFINED)
[build] /Users/julianuspfeuffer/git/OpenMS/src/openms/thirdparty/percolator/ScoreHolder.h:51:45: warning: field 'label' will be initialized after field 'pPSM' [-Wreorder-ctor]
[build]    51 |       : score(s), q(0.0), pep(0.0), p(0.0), label(l), pPSM(psm) {}
[build]       |                                             ^~~~~~~~  ~~~~~~~~~
[build]       |                                             pPSM(psm) label(l)
[build] In file included from /Users/julianuspfeuffer/git/OpenMS/src/openms/thirdparty/percolator/Scores.cpp:37:
[build] /Users/julianuspfeuffer/git/OpenMS/src/openms/thirdparty/percolator/SetHandler.h:95:13: warning: 'const' type qualifier on return type has no effect [-Wignored-qualifiers]
[build]    95 |   LabelType const getLabel(int setPos);
[build]       |             ^~~~~
[build] 3 warnings generated.
```

Either fix them by patch or upstream commit or put percolator behind -isystem or otherwise silence its warnings.

--- a/src/openms/CMakeLists.txt
+++ b/src/openms/CMakeLists.txt
@@ -71,7 +71,7 @@ if (MSVC)
 else()
   set_source_files_properties(${OpenMS_percolator_sources}
     PROPERTIES
-      COMPILE_FLAGS "-Wno-sign-compare -Wno-unused-parameter -Wno-deprecated-declarations -Wno-unused-variable -Wno-overloaded-virtual")
+      COMPILE_FLAGS "-Wno-sign-compare -Wno-unused-parameter -Wno-deprecated-declarations -Wno-unused-variable -Wno-overloaded-virtual -Wno-reorder -Wno-reorder-ctor -Wno-uninitialized -Wno-ignored-qualifiers -Wno-unused-but-set-variable")
   set_source_files_properties(${OpenMS_percolator_sources}
     PROPERTIES COMPILE_OPTIONS "-fvisibility=hidden")
 endif()

